### ISSUE TITLE
Please add the fact that WSL2 is not co-exists with VirtualBox 6.0.x

### ISSUE THREAD

**Issue by rengui** [MicrosoftDocs/WSL#536]
wsl2-faq [Will I be able to run WSL 2 and other 3rd party virtualization tools such as VMware, or VirtualBox?](https://docs.microsoft.com/en-us/windows/wsl/wsl2-faq#will-i-be-able-to-run-wsl-2-and-other-3rd-party-virtualization-tools-such-as-vmware-or-virtualbox) states:
> ......and versions of VirtualBox before VirtualBox 6 (VirtualBox 6.0.0 released in December 2018 now supports Hyper-V as a fallback execution core on a Windows host!)

This is quite misleading...

according to [here](https://github.com/microsoft/WSL/issues/4174) and [here](https://forums.virtualbox.org/viewtopic.php?f=6&t=90853&start=60#p445491), WSL2 is still not co-exists with VirtualBox 6.0.x yet.

when WSL2 is running, starting VirtualBox 6.0.x virtual machine will failed: 
> Call to WHvSetupPartition failed: ERROR_SUCCESS (Last=0xc000000d/87) (VERR_NEM_VM_CREATE_FAILED).
Result Code: | E_FAIL (0x80004005)
-- | --
Component: | ConsoleWrap
Interface: | IConsole {872da645-4a9b-1727-bee2-5585105b9eed}

I'm quite expecting WSL2 and VirtualBox 6.0 can running at the same time, but before that comes true, could you please correct the faq doc? 
Thanks!

**Comment by craigloewen-msft** [comment]
From the [Virtual Box 6.0.0 changelog](https://www.virtualbox.org/wiki/Changelog-6.0), it states that it can also use Hyper-V as the fallback hypervisor.

![image](https://user-images.githubusercontent.com/42221804/71109800-dca8d400-217a-11ea-9e10-dc8ec022dbed.png)

As well, [Oracle's documentation says](https://docs.oracle.com/cd/E97728_01/F12469/html/hyperv-support.html) that it's possible to run Hyper-V with Virtual Box. 

I think you're running into a specific error, to me Oracle is clearly indicating that they do have support for Hyper-V which is why we have the link our docs that say so. Is there a way we could make the statement clearer?

**Comment by rengui** [comment]
yes, Hyper-V and VirtualBox 6.0 looks can play together, as the Oracle document you mentioned as well as the  [here](https://forums.virtualbox.org/viewtopic.php?f=6&t=90853&start=60) (those time there is no WSL2 yet).


but now it really looks problem to have WSL2 together with VirtualBox, 

-   [Are WSL 2 and VirtualBox currently mutually exclusive?](https://forums.virtualbox.org/viewtopic.php?f=6&t=95426)

    >    Yes, WSL2 is not compatible with Virtualbox, due to WSL2 using Hyper-V, **which uses VT-x exclusively** and doesn't share it with Virtualbox. To use Virtualbox properly, for now*, you have to have Hyper-V off, which turns off anything that uses Hyper-V. 

-   [VirtualBox 6.0 and Hyper-V]( https://forums.virtualbox.org/viewtopic.php?f=6&t=90853&start=195#p465333)

    >    Hyper-V usage by VirtualBox started working with 1809, and **some change in 1903 broke it**. 
    >    ......
    >    The 6.1.0_BETA1 seemingly fixed it (but in fact it just accidentally disabled the optimization, so it was REALLY slow), and 6.1.0_RC1 restored the optimization which means that it doesn't work again with 1903 and 1909. 

- [Hyper-V Powering Windows Features (2019.12.12)](https://techcommunity.microsoft.com/t5/Virtualization/Hyper-V-Powering-Windows-Features/ba-p/1063685)
    >    Unless I missed something, it isn't. If you enable even Sandbox (which uses only one Hyper-V service), for example, Virtualbox doesn't run. 

<br/>
<br/>
<br/>
 maybe it's not a bug of `WSL doc` nor WSL2,  but Windows (Hyper-V API) or VirtualBox......
<br/>
<br/>
Just record some more details for the errors I met. (maybe there have no way but waiting newer version of Windows10 and/or VirtualBox)

**HostOS**:  Win10 Insider 18990 x64  (for WSL2)
**VirtualBox**: latest 6.0.14  (as of now 6.1.0 comes out, maybe need to try on it.)

Errors when starting VMs when Hypver-V enabled (with WSL2 running)  -- **All VMs are able to running correctly when disabled Hyper-V.**

- uncheck "Enable VT-x/AMD-v" (32bit Guest OS)
    ERROR shows on GUI:

    ```
    Raw-mode is unavailable courtesy of Hyper-V. (VERR_SUPDRV_NO_RAW_MODE_HYPER_V_ROOT).
    ```

    VM log:

    ```
    CPUM: No hardware-virtualization capability detected
    ```

    

- "Enable VT-x/AMD-v" (32-bit Guest OS; or 64-bit Guest OS which must check this)
    ERROR shows on GUI:

    ```
    Call to WHvSetupPartition failed: ERROR_SUCCESS (Last=0xc000000d/87) (VERR_NEM_VM_CREATE_FAILED).
    ```

    VM log 

    - 32-bit Guest OS: 

        ```
        00:00:01.795686 HM: HMR3Init: Attempting fall back to NEM: VT-x is not available
        00:00:01.816332 CPUM: No hardware-virtualization capability detected
        ```

    - 64-bit Guest OS

        ```
        00:00:01.824889 HM: HMR3Init: Attempting fall back to NEM: VT-x is not available
        00:00:01.844124 NEM: Adjusting APIC configuration from X2APIC to APIC max mode.  X2APIC is not supported by the WinHvPlatform API!
        00:00:01.844126 NEM: Disable Hyper-V if you need X2APIC for your guests!
        ```

**Comment by Biswa96** [comment]
Source: https://forums.virtualbox.org/viewtopic.php?f=6&t=90853&sid=f20b0f7f51e462005f9b3be80a567617&start=120#p453078

Try this command:

    VBoxManage setextradata global "VBoxInternal/NEM/UseRing0Runloop" 0

It works but very slow with type 2 hypervisor because type 1 is already in the chair.

**Comment by rengui** [comment]
> Source: https://forums.virtualbox.org/viewtopic.php?f=6&t=90853&sid=f20b0f7f51e462005f9b3be80a567617&start=120#p453078
> 
> Try this command:
> 
> ```
> VBoxManage setextradata global "VBoxInternal/NEM/UseRing0Runloop" 0
> ```
> 
> It works but very slow with type 2 hypervisor because type 1 is already in the chair.

Thanks a lot. It really works :) 

follow the same link, below is a bit safer to try(affect not globally, but only the VM you are editing)
```
# add one line to your VM.box file <ExtraData> section
<ExtraDataItem name="VBoxInternal/NEM/UseRing0Runloop" value="0"/>
```

**Comment by rengui** [comment]
Thanks @Biswa96 

Since the WA 
```
goto C:\Program Files\Oracle\VirtualBox
# for one VM
VBoxManage setextradata "<VM Name>" "VBoxInternal/NEM/UseRing0Runloop" 0
or 
# globally
VBoxManage setextradata global "VBoxInternal/NEM/UseRing0Runloop" 0
```
is working with 
- Win10 Insider 18990 x64 (for WSL2)
- VirtualBox-6.1.0

I'm closing this ticket.

**Comment by wooseopkim** [comment]
I've struggled for days because of this! Thanks for all the work. You saved my day.

**Comment by Paxhape** [comment]
Same problem here, thanks very much for helping ! 
Is there any efect on performance of VM ?

**Comment by earonesty** [comment]
Yes, running with this disabled hurts the performance of vbox.  WSL consumes some sort of "hypervisor type 1 slot", so vbox won't run.   You *should* be able to run both at once, but you can't.  Shutting down WSL2, and starting VBOX can be done on the same machine.... but not both at once.

**Comment by rahmi** [comment]
Currenty at Virtualbox 6.1.8, converting WSL's ubuntu-20.04 to WSL2 on Windows 10 2004 (May 10th.) Enables Windows Virtualization Platform.

As a virtualbox guest I am running Manjaro on kernel version 5.6, this works flawlessly along with other distributions as well, but when enabling Windows Virtualization Platform, what happens is that there seem to be some memory segmentation problems, especially related to cryptographic tools like PGP. One example is: updating packages that are gpg signed, verification of the signature and package integrity fails.

This nature is similar on all the virtualbox guests i have tested, like Debian, Manjaro, Fedora e.t.c..
I tested on several because I had to eliminate any flawed nature of the manjaro and archlinux keyrings that are cached.

Rolling the ubuntu instance back to WLS1 and turning the virtualization feature of in Windows 10 pro, returns virtualbox into a working state.

I have therefore concluded that running WSL2 and Virtualbox is not possible as it is.
I am instead running Visual studio code, connected to ssh remote (vscode extention) on my manjaro instance + docker setup (manjaro is running a vscode server to provide connectivity over ssh, and this works just as well or maybe slightly better performance wise than connecting vscode to a WSL2 instance running docker for example.

**Comment by gitthail** [comment]
I don't think I understand the purpose WSL2. You can't break all virtualization, that renders it useless to me.

I think the solution (when I can face the effort) is uninstall windows, put fedora on the bare metal and then kvm/qemu windows on top with hardware passthrough, and any other VMs needed can run in parallel.

**Comment by oki-burokku** [comment]
WSL2 is not yet fit for purpose. Wasted best part of a day trying to figure out this one. Winding back to WSL1.

**Comment by jlearman** [comment]
I too am unable to get WSL2 to work with VirtualBox.  I'm running Ubuntu on both WSL and VB.  When WSL2 is enabled and I start my Ubuntu in VB, I get a blank rectangular screen with a rapidly blinking cursor, which persists indefinitely.  Of course, I configured HyperV for the VB VM, and made sure that HyperV was enabled in Windows Features.

What's worse, disabling WSL2 using any of the normal means doesn't restore VB to working state.  Fortunately I was able to use Windows Restore to just before I installed WSL2 and it works.  I did this twice; the second time I had to restore back to the first installation of WSL2, and this time I also had to reinstall ("Repair") VB.

It would be great to use WSL2, but for now I have to stick to WSL1 because I need VirtualBox; I need its ability to checkpoint.

**Comment by oki-burokku** [comment]
WSL2 works fine with vmware player (which is the setup I am using now). Microsoft broke something in virtualbox with WSL2, I have verified this on a couple different workstations.  Did not appear to be virtualbox's fault, but it is unusable ATM. I guess it will be fixed sometime....

**Comment by goigle** [comment]
I'm on an AMD platform (Ryzen 7 2700x). VirtualBox 6.1.14 did not work for me, but it seems 6.1.16 is working now.
For Windows features, I have neither Hyper-V nor Windows Hypervisor Platform installed. The Virtual Machine Platform and Windows Subsystem for Linux are installed.

Before 6.1.16 I could barely get the Live CD for Ubuntu opened and it crashed at install, now it's actually installing.

**Comment by ivictbor** [comment]
> Thanks @Biswa96
> 
> Since the WA
> 
> ```
> goto C:\Program Files\Oracle\VirtualBox
> # for one VM
> VBoxManage setextradata "<VM Name>" "VBoxInternal/NEM/UseRing0Runloop" 0
> or 
> # globally
> VBoxManage setextradata global "VBoxInternal/NEM/UseRing0Runloop" 0
> ```
> 
> is working with
> 
> * Win10 Insider 18990 x64 (for WSL2)
> * VirtualBox-6.1.0
> 
> I'm closing this ticket.

Works but. Machine was loaded, worked several minutes and got freezed. Probably just because of slow hypervisor type 2 performance

**Comment by onomatopellan** [comment]
@ivictbor Try latest version 6.1.16, it has some fixes for Hyper-V.
https://www.virtualbox.org/wiki/Downloads

**Comment by JulioJu** [comment]
@onomatopellan but it could work, or not . See https://github.com/MicrosoftDocs/WSL/issues/798

**Comment by Spongman** [comment]
this should be reopened. the workarounds only work for some users and even when they do the performance is terrible and the guests are still unstable and prone to crashes. yes, with VB 6.1.16

**Comment by pabloruizsoria** [comment]
It is not working here. I have wasted 2 days before reading this topic.
Windows 10 pro
WSL installed 2 days ago
Virtual Box and extension pack reinstalled twice in the last version

I cannot install new machines
I cannot run machines previously installed

**Comment by mareksloser** [comment]
> It is not working here. I have wasted 2 days before reading this topic.
> Windows 10 pro
> WSL installed 2 days ago
> Virtual Box and extension pack reinstalled twice in the last version
> 
> I cannot install new machines
> I cannot run machines previously installed

I have same situation. I cant use docker and VirtualBox at same time. When i switch off windows function hyper-V, i can use Virtualbox, but cant use a docker.. (it can solve your problem)

**Comment by Salmondx** [comment]
@craigloewen-msft could you please provide your vision if this will ever be solved for WSL2? Currently we can't use WSL2 at all because it blocks us from using vagrant and virtualbox, and since WSL1 is no longer supported - news are bad for web developers.

That's a really common question when people are switching from WSL1 to WSL2 (e.g. lots of themes on reddit, stackoverflow and etc, this one for example: https://stackoverflow.com/questions/62184881/virtualbox-and-wsl2).

Vagrant/Virtualbox pair are still popular especially among web developers (e.g. its de facto a standard dev env for Laravel and other web frameworks).

**Comment by craigloewen-msft** [comment]
@Salmondx [Virtual Box 6.0.0](https://www.virtualbox.org/wiki/Changelog-6.0) and [VMware 15.5](https://blogs.vmware.com/workstation/2020/05/vmware-workstation-now-supports-hyper-v-mode.html) both support running simultaneously with Hyper-V. The Hyper-V team has been working with these other tools to ensure that they can run well with Hyper-V enabled. I understand that in cases like VirtualBox performance can be degraded right now, but to answer the base question of this thread, yes they are currently able to exist together, and I personally would hope that the performance here would improve so both can be run simultaneously with no perceived degradation.

**Comment by Salmondx** [comment]
Thanks for response! The problem is that it's not even possible to mount folders to virtualbox Hyper-V VM from WSL2 filesystem. May be in some artificial example they can coexist, but it's far from being useful. Given that files should be stored inside WSL2 filesystem now, it's safe to say that they don't work together.

**Comment by craigloewen-msft** [comment]
I see, is the feature request here that you can use both tools simultaneously for separate projects? Or that you'd like to use both tools simultaneously on the same projects (i.e: Store your project in the WSL 2 file system and then access that same project folder from a VirtualBox VM?)

**Comment by Salmondx** [comment]
@craigloewen-msft on the same project. Imagine a typical nodejs/laravel/whatever web app: it should live in WSL2 filesystem so you could use all the required tools like npm, compilers and etc. And virtualbox/vagrant should be able to mount folders from WSL2 filesystem into VM.

That's how I use WSL1 today (and probably almost everyone in this thread 😄).

**Comment by Cito** [comment]
@Salmondx It's even a problem when you want to use them in different projects, because every time you need VirtualBox you need to first un-install the WSL feature which is extremely inconvenient, since it also needs a reboot.

@craigloewen-msft No, they are not able to exist together. It's not only that if you have WSL-2 installed, then the performance is slower. It's also that (depending on your CPU and the guest BS) some virtual boxes simply crash which work otherwise well when WSL-2 is not installed. It's not only me, many other users are reporting this in #798.

**Comment by Spongman** [comment]
> in cases like VirtualBox performance can be degraded right now,

@craigloewen-msft i think you're understating the issue here. the performance, when it works at all, is terrible.

maybe there _is_ some hidden communication going on between MS & Oracle, but all i see is bidirectional finger-pointing and/or whitewashing.

**Comment by ivictbor** [comment]
> @Salmondx It's even a problem when you want to use them in different projects, because every time you need VirtualBox you need to first un-install the WSL feature which is extremely inconvenient, since it also needs a reboot.
> 
> @craigloewen-msft No, they are not able to exist together. It's not only that if you have WSL-2 installed, then the performance is slower. It's also that (depending on your CPU and the guest BS) some virtual boxes simply crash which work otherwise well when WSL-2 is not installed. It's not only me, many other users are reporting this in #798.

I am doing exactly like this, 

1. When I need VM, I run

```
DISM /Online /Disable-Feature:VirtualMachinePlatform PowerShell Disable-WindowsOptionalFeature -Online -FeatureName VirtualMachinePlatform -All bcdedit /set hypervisorlaunchtype off
```

2. Reboot

3. Work in VM

4.

```
DISM /Online /Enable-Feature:VirtualMachinePlatform PowerShell Enable-WindowsOptionalFeature -Online -FeatureName VirtualMachinePlatform -All bcdedit /set hypervisorlaunchtype auto start
```

5. Reboot back


I was asked about it too many times, so I even created a hint to paste URL😀  https://hinty.io/devforth/wsl-2-starter-guide-must-read/


Yes WSL 2 is cool, but seams like for now it is easier to buy another PC if you need a VirtualMachines for development

**Comment by Cito** [comment]
@Spongman The problems seem to be hard to reproduce and depend on things like the host CPU, the guest OS and the various settings in VB. I have the impression that the performance degradation and frequency of crashes seem to be worse for some than for others. That's probably one of the reasons why some believe it already works and this is not really moving forward.

**Comment by zsbaksa** [comment]
> > @Salmondx It's even a problem when you want to use them in different projects, because every time you need VirtualBox you need to first un-install the WSL feature which is extremely inconvenient, since it also needs a reboot.
> > @craigloewen-msft No, they are not able to exist together. It's not only that if you have WSL-2 installed, then the performance is slower. It's also that (depending on your CPU and the guest BS) some virtual boxes simply crash which work otherwise well when WSL-2 is not installed. It's not only me, many other users are reporting this in #798.
> 
> I am doing exactly like this,
> 
> 1. When I need VM, I run
> 
> ```
> DISM /Online /Disable-Feature:VirtualMachinePlatform 
> PowerShell Disable-WindowsOptionalFeature -Online -FeatureName VirtualMachinePlatform -All 
> bcdedit /set hypervisorlaunchtype off
> ```
> 
> 1. Reboot
> 2. Work in VM
> 
> ```
> DISM /Online /Enable-Feature:VirtualMachinePlatform 
> PowerShell Enable-WindowsOptionalFeature -Online -FeatureName VirtualMachinePlatform -All 
> bcdedit /set hypervisorlaunchtype auto start
> ```
> 
> 1. Reboot back
> 
> I was asked about it too many times, so I even created a hint to paste URL😀 https://hinty.io/devforth/wsl-2-starter-guide-must-read/
> 
> Yes WSL 2 is cool, but seams like for now it is easier to buy another PC if you need a VirtualMachines for development

Mate, you just saved my whole day. My VM now is working again, without any issues :)

**Comment by Pysis868** [comment]
Rebooting is not an option.  Clearly WSL2/Windows VM functionality is causing the conflicts here and even with VirtualBox 6.1.18 and becoming exhausted with settings and lengthy guides this situation still needs to improve.
Like others said, I will downgrade WSL to 1 or disable entirely and find another way to run Docker, windows subsystem support, Desktop app for it, or even neither....

**Comment by gitthail** [comment]
I've taken a swing at this on 3 separate occasions over the past 2 years. 

Don't waste your time. 

If you need VBox for work, don't even look at WSL

**Comment by rahmi** [comment]
> I've taken a swing at this on 3 separate occasions over the past 2 years.
> 
> Don't waste your time.
> 
> If you need VBox for work, don't even look at WSL

For the salvation of my own sanity I installed Manjaro Linux and my problems are long gone.
No point in trying to make Windows something it is not.

**Comment by peschue** [comment]
VirtualBox 6.1.22 does not work with the `VBoxManage setextradata global "VBoxInternal/NEM/UseRing0Runloop" 0` described above.

Attaching the log of my attempt. [VBox.log](https://github.com/MicrosoftDocs/WSL/files/6675230/VBox.log)


Error is
```
00:00:02.790862 VMSetError: F:\tinderbox\win-6.1\src\VBox\VMM\VMMR3\NEMR3Native-win.cpp(1418) int __cdecl nemR3NativeInitAfterCPUM(struct VM *); rc=VERR_NEM_VM_CREATE_FAILED
00:00:02.790891 VMSetError: Failed to get device handle and/or partition ID for 0000000001353ac0 (hPartitionDevice=0000000000000c45, Last=0xc0000002/1)
00:00:02.790907 NEM: Destroying partition 0000000001353ac0 with its 0 VCpus...
00:00:02.792421 ERROR [COM]: aRC=E_FAIL (0x80004005) aIID={872da645-4a9b-1727-bee2-5585105b9eed} aComponent={ConsoleWrap} aText={Failed to get device handle and/or partition ID for 0000000001353ac0 (hPartitionDevice=0000000000000c45, Last=0xc0000002/1) (VERR_NEM_VM_CREATE_FAILED)}, preserve=false aResultDetail=-6805
00:00:02.792580 Console: Machine state changed to 'PoweredOff'
00:00:02.813113 Power up failed (vrc=VERR_NEM_VM_CREATE_FAILED, rc=E_FAIL (0X80004005))
00:00:03.313816 GUI: UIMachineViewNormal::resendSizeHint: Restoring guest size-hint for screen 0 to 800x600
00:00:03.313907 ERROR [COM]: aRC=E_ACCESSDENIED (0x80070005) aIID={4680b2de-8690-11e9-b83d-5719e53cf1de} aComponent={DisplayWrap} aText={The console is not powered up}, preserve=false aResultDetail=0
00:00:03.314591 GUI: Aborting startup due to power up progress issue detected...
```

Tried many combinations, and I have all my development in WSL2, so will probably need to drop VirtualBox and use a separate machine for embedded programming where I need USB/JTAG access from Linux.

**Comment by Faq** [comment]
@peschue is it working with vb 6.1.24? also better to oen new issue (if there is not one) as this one is closed.

**Comment by Agrover112** [comment]
Actually I agree VirtualBox doesn't work

**Comment by ivictbor** [comment]
Actually we all must admit that it never worked. I mean that temporary fix made vbox awfully slow anyway so I doubt that any guest os with minimal X server was usable by anyone, if I am wrong could someone please share expirience?
And disabling hyperv and rebooting any time when you need to work on other distro/image is definitely not a way to go

**Comment by gitthail** [comment]
It never worked for people who require performance from the guest. I've gone back to windows and tried to get it going several times over the years.

**Comment by Pehrsonal** [comment]
had been lovely to have virtualbox and sub-system running at the same time!

**Comment by OpenCoderX** [comment]
Just dropping by to express sadness.

**Comment by Agrover112** [comment]
@OpenCoderX  use VMWare?

**Comment by CTBlaze** [comment]
It still doesn't work, WIn11 Home, WSL2, latest updates and
VirtualBox 6.1.32 (and lower versions) 
Whatever happened to virtualization being a total abstraction and indirection for the guest/client? 
Why does Hyper-V have to reserve the ACTUAL hardware exclusively?  this is just wrong, Microsoft, and once again I say: you should be sued by the world's people for havin introduced and caused so much inproductivity and loss of energy and time, that you should be shut down for good ;)

**Comment by CTBlaze** [comment]
having said that, anybody tried running VirtualBox inside Ubuntu which runs in WSL2 ?

**Comment by gmankab** [comment]
still don't work

**Comment by norru** [comment]
As of 6.1.32, it's still impossible to run WSL2 and VirtualBox at the same time.

**Comment by Cito** [comment]
> As of VB 6.1.32, it's still impossible to run WSL2 and VirtualBox at the same time.

According to my own experience and what I read from the comments above, it works, with the caveat that VirtualBox is slower ("turtle mode"). How much slower depends on the CPU and configuration of the VM, e.g. some claim that VMs with only one CPU assigned work better. Some also report crashes, but this seems to happen only with certain host CPUs and VM configurations, and should be reported on the VirtualBox bug tracker.

**Comment by norru** [comment]
Works

> > As of VB 6.1.32, it's still impossible to run WSL2 and VirtualBox at the same time.
> 
> According to my own experience and what I read from the comments above, it works, with the caveat that VirtualBox is slower ("turtle mode"). How much slower depends on the CPU and configuration of the VM, e.g. some claim that VMs with only one CPU assigned work better. 

I've tried that, it's unusable for me as I'm running a desktop in the VM. Hence "doesn't work".

> Some also report crashes, but this seems to happen only with certain host CPUs and VM configurations, and should be reported on the VirtualBox bug tracker.

Yes, I think it's a matter for the VB dev team now.

**Comment by lx30011** [comment]
what a mess

**Comment by mims92** [comment]
I have virtualbox 6.1.34 and wsl2 and it is working. I had to change the acceleration (system >acceleration)  from default to Hyper-v.

**Comment by norru** [comment]
> I have virtualbox 6.1.34 and wsl2 and it is working. I had to change the acceleration (system >acceleration) from default to Hyper-v.

Doesn't work for me if I use more than 1 CPU in the VM.

**Comment by mims92** [comment]
True, i have updated my comment

**Comment by dbjungle** [comment]
> > I have virtualbox 6.1.34 and wsl2 and it is working. I had to change the acceleration (system >acceleration) from default to Hyper-v.
> 
> Doesn't work for me if I use more than 1 CPU in the VM. The VM just doesn't boot (or takes so long to boot that it's indistinguishable from a crash)

I have the exact same experience.  I have seen a lot of reports suggesting they can work together, but this is the only time I've seen it mentioned that it only works with one CPU assigned.  I am now able to boot my VBox VM, but I wouldn't call this working.

**Comment by AlexMakesSoftware** [comment]
Why is this closed? This is still an issue. I wasted most of a day trying to get a VM working only to discover the cause was my WSL installation. WSL is useless to me if I can't run my virtual machines too. I got around this problem by uninstalling the WSL and turning off hyper and virtualisation software and just using VirtualBox instead.

Reopen please.

**Comment by CTBlaze** [comment]
> Just stumbled upon this bug, I wonder why it's closed since it is still present (and made me waste an entire afternoon).

Because Microsoft and their support chain are ignorant and arrogant. Also, there is a virtualization war going on. They trying to keep the competition out! (Hyper-V reserves the _actual_ HW resources almost exclusively, instead of abstracting it all and share resource pool. LEAVE IT TO THE USERS, M$ !

**Comment by abdarafi** [comment]
will there be any clarification of why it's closed? c'mon, I know you guys are brilliant. please consider being open to us as part of the community. a little hint would be sufficient and helpful.

**Comment by chylex** [comment]
> Doesn't work for me if I use more than 1 CPU in the VM. The VM just doesn't boot (or takes so long to boot that it's indistinguishable from a crash)

I would have never guessed to try that, but I got Fedora to finally boot with 1 CPU and it was usable enough for me. However, a Windows VM is completely unusable.

If this doesn't get fixed, guess I'll have to remove WSL2 and Hyper-V from my machines and go back to running Docker in a Linux VM...

**Comment by daveloyall** [comment]
FYI, I was just able to use my guest VM on Windows 10 (1889) under Virtualbox 6.1.36 and it didn't crash when I updated packages inside it.  I'm using the "turtle mode" (NEM) because VT-x is not available because Hyper-V is active.

Prior to 6.1.36, updating packages in the VM would cause it to crash around the time messages like this would be spammed to the guest log many times:

```
02:34:31.184474 TM: Giving up catch-up attempt at a 60 004 978 694 ns lag; new total: 4 740 349 480 465 ns
```

Please note the specific versions I named here and the date of this post.  We've been warned by a Virtualbox forum moderator that things may change:

> If Hyper-V cannot be disabled, e.g. due to company policy, then VirtualBox v6.x has a new ability to run via the Windows Hypervisor Platform (NEM), however Windows 10 (1809) or later is required, the feature still has bugs (as of VirtualBox 6.1.12), Microsoft keeps changing it, and it currently causes a significant drop in VM performance.

Good luck!

**Comment by internationils** [comment]
Update from May 2023...
I just installed kubuntu 23.04 under win11 with VirtualBox 7.0.8 and it seems to work at normal speed. Currently only using 1 CPU though, maybe as per a few comments up it will slow down with more CPUs.

**Comment by irryden** [comment]
Similar situation here. Was running Virtualbox 6.1.26 and got stuck with VMs that failed to start after installing WSL2/Ubuntu. Just an awful user experience, for something that has been known about for years apparently. After installing VirtualBox 7.0.8 (latest as of this writing) the VMs could start again. Have not measured performance.

**Comment by Kazmirchuk** [comment]
I'm still having problems with "rcu_preempt detected stalls" in VirtualBox 7.1.8 and WSL version 2.4.13.0
